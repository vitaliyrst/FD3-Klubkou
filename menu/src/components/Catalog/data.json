{
  "name": "Catalog",
  "type": "CATEGORY",
  "children": [
    {
      "name": "folder1",
      "type": "CATEGORY",
      "children": []
    },
    {
      "name": "folder2",
      "type": "CATEGORY",
      "children": [
        {
          "name": "folder21",
          "type": "CATEGORY",
          "children": [
            {
              "name": "file211",
              "type": "ITEM"
            },
            {
              "name": "file212",
              "type": "ITEM"
            },
            {
              "name": "file213",
              "type": "ITEM"
            }
          ]
        },
        {
          "name": "folder22",
          "type": "CATEGORY",
          "children": []
        }
      ]
    },
    {
      "name": "folder3",
      "type": "CATEGORY",
      "children": [
        {
          "name": "file31",
          "type": "ITEM"
        }
      ]
    }
  ]
}