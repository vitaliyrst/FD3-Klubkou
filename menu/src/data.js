const data = {
    "name": "Каталог",
    "type": "CATEGORY",
    "children": [
        {
            "name": "Романы",
            "type": "CATEGORY",
            "children": []
        },
        {
            "name": "Бестселлеры",
            "type": "CATEGORY",
            "children": [
                {
                    "name": "Приключения",
                    "type": "CATEGORY",
                    "children": [
                        {
                            "name": "Инферно.pdf",
                            "type": "ITEM"
                        },
                        {
                            "name": "Ангелы и демоны.pdf",
                            "type": "ITEM"
                        },
                        {
                            "name": "Утраченный символ.pdf",
                            "type": "ITEM"
                        }
                    ]
                },
                {
                    "name": "Детективы",
                    "type": "CATEGORY",
                    "children": []
                }
            ]
        },
        {
            "name": "Фантастика",
            "type": "CATEGORY",
            "children": [
                {
                    "name": "Фентези",
                    "type": "CATEGORY",
                    "children": [
                        {
                            "name": "Властелин колец.pdf",
                            "type": "ITEM"
                        }
                    ]
                },
                {
                    "name": "Научная фантастика",
                    "type": "CATEGORY",
                    "children": [
                        {
                            "name": "Солярис.pdf",
                            "type": "ITEM"
                        }
                    ]
                }
            ]
        }
    ]
}

export default data;