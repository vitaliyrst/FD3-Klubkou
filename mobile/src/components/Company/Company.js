import React from 'react';
import './Company.css';
import Client from "../Clients/Client";
import CompanyEvent from "../EE";
import ClientForm from "../Clients/ClientForm";

class Company extends React.PureComponent {
    state = {
        clients: this.props.clients,
        name: this.props.name,
        status: 'all',
        form: null,
        client: {}
    }

    componentDidMount = () => {
        CompanyEvent.addListener('edit', this.handleEdit);
        CompanyEvent.addListener('delete', this.handleDelete);
        CompanyEvent.addListener('save', this.handleSave);
        CompanyEvent.addListener('cancel', this.handleCancel);
    }

    handleEdit = (client) => this.setState({form: 'edit', client: client});

    handleDelete = (id) => this.setState({clients: this.state.clients.filter(client => client.id !== id), form: null});

    handleSave = (data) => {
        if (this.state.form === 'add') {
            data.id = this.getMaxId();
            const newClients = [...this.state.clients, data];
            this.setState({clients: newClients, form: null});
        } else {
            const editedClients = this.state.clients.map(client => (client.id === data.id) ? data : client);
            this.setState({clients: editedClients, form: null, client: {}});
        }
    }

    getMaxId = () => {
        let max = 0;
        this.state.clients.forEach(item => (item.id >= max) ? max = item.id : null);
        return max += 1;
    }

    handleCancel = () => this.setState({form: null, client: {}});

    handleSetCompany = (name) => this.setState({name});

    handleSetStatus = (status) => this.setState({status});

    getClients = () => {
        if (this.state.status === 'all') {
            return this.state.clients.map(client => <Client key={client.id} client={client}/>);
        }
        return this.state.clients.map(client => {
            return (this.state.status === client.status) ? <Client key={client.id} client={client}/> : null;
        });
    }

    render() {
        console.log(`Company ${this.state.name} render`);

        return (
            <div>
                <div>
                    <input type='button' value='Velcom' onClick={() => this.handleSetCompany('Velcom')}/>
                    <input type='button' value='MTC' onClick={() => this.handleSetCompany('MTC')}/>
                    <h2>Компания: {this.state.name}</h2>
                </div>
                <div className='filter'>
                    <input type='button' value='Все' onClick={() => this.handleSetStatus('all')}/>
                    <input  type='button' value='Активные' onClick={() => this.handleSetStatus('active')}/>
                    <input  type='button' value='Заблокированные' onClick={() => this.handleSetStatus('blocked')}/>
                </div>
                <div>
                    <table>
                        <tbody>
                        <tr>
                            <th>Фамилия</th>
                            <th>Имя</th>
                            <th>Отчество</th>
                            <th>Баланс</th>
                            <th>Статус</th>
                            <th>Редактировать</th>
                            <th>Удалить</th>
                        </tr>
                        {this.getClients()}
                        </tbody>
                    </table>
                </div>
                <input type='button' value='Добавить' onClick={() => this.setState({form: 'add', client: {}})}/>
                <div>
                    {this.state.form && <ClientForm client={this.state.client}/>}
                </div>
            </div>
        );
    }

}

export default Company;