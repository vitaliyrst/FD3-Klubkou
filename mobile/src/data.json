[
  {
    "id": 1,
    "lastName": "Иванов",
    "firstName": "Иван",
    "middleName": "Иванович",
    "balance": 200,
    "status": "active"
  },
  {
    "id": 2,
    "lastName": "Сидоров",
    "firstName": "Сидор",
    "middleName": "Сидорович",
    "balance": 250,
    "status": "active"
  },
  {
    "id": 3,
    "lastName": "Петров",
    "firstName": "Петр",
    "middleName": "Петрович",
    "balance": 180,
    "status": "active"
  },
  {
    "id": 4,
    "lastName": "Григорьев",
    "firstName": "Григорий",
    "middleName": "Григорьевич",
    "balance": -220,
    "status": "blocked"
  }
]