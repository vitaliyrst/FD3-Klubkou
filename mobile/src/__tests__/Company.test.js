import React from 'react';
import renderer from 'react-test-renderer';
import Company from "../components/Company/Company";
import data from '../data.json';

test('company', () => {
    const component = renderer.create(
        <Company clients={data} name='Velcom'/>
    );

    let three = component.toJSON();
    expect(three).toMatchSnapshot();

    const filter = component.root.find(input => input.props.className === 'filter').children;
    filter.forEach(input => {
        input.props.onClick();
        three = component.toJSON();
        expect(three).toMatchSnapshot();
    })
});