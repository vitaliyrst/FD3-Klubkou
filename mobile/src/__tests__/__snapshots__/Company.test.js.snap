// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`company 1`] = `
<div>
  <div>
    <input
      onClick={[Function]}
      type="button"
      value="Velcom"
    />
    <input
      onClick={[Function]}
      type="button"
      value="MTC"
    />
    <h2>
      Компания: 
      Velcom
    </h2>
  </div>
  <div
    className="filter"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <div>
    <table>
      <tbody>
        <tr>
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
        <tr>
          <td>
            Иванов
          </td>
          <td>
            Иван
          </td>
          <td>
            Иванович
          </td>
          <td>
            200
          </td>
          <td
            style={
              Object {
                "backgroundColor": "green",
              }
            }
          >
            active
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr>
          <td>
            Сидоров
          </td>
          <td>
            Сидор
          </td>
          <td>
            Сидорович
          </td>
          <td>
            250
          </td>
          <td
            style={
              Object {
                "backgroundColor": "green",
              }
            }
          >
            active
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr>
          <td>
            Петров
          </td>
          <td>
            Петр
          </td>
          <td>
            Петрович
          </td>
          <td>
            180
          </td>
          <td
            style={
              Object {
                "backgroundColor": "green",
              }
            }
          >
            active
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr>
          <td>
            Григорьев
          </td>
          <td>
            Григорий
          </td>
          <td>
            Григорьевич
          </td>
          <td>
            -220
          </td>
          <td
            style={
              Object {
                "backgroundColor": "red",
              }
            }
          >
            blocked
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить"
  />
  <div />
</div>
`;

exports[`company 2`] = `
<div>
  <div>
    <input
      onClick={[Function]}
      type="button"
      value="Velcom"
    />
    <input
      onClick={[Function]}
      type="button"
      value="MTC"
    />
    <h2>
      Компания: 
      Velcom
    </h2>
  </div>
  <div
    className="filter"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <div>
    <table>
      <tbody>
        <tr>
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
        <tr>
          <td>
            Иванов
          </td>
          <td>
            Иван
          </td>
          <td>
            Иванович
          </td>
          <td>
            200
          </td>
          <td
            style={
              Object {
                "backgroundColor": "green",
              }
            }
          >
            active
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr>
          <td>
            Сидоров
          </td>
          <td>
            Сидор
          </td>
          <td>
            Сидорович
          </td>
          <td>
            250
          </td>
          <td
            style={
              Object {
                "backgroundColor": "green",
              }
            }
          >
            active
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr>
          <td>
            Петров
          </td>
          <td>
            Петр
          </td>
          <td>
            Петрович
          </td>
          <td>
            180
          </td>
          <td
            style={
              Object {
                "backgroundColor": "green",
              }
            }
          >
            active
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr>
          <td>
            Григорьев
          </td>
          <td>
            Григорий
          </td>
          <td>
            Григорьевич
          </td>
          <td>
            -220
          </td>
          <td
            style={
              Object {
                "backgroundColor": "red",
              }
            }
          >
            blocked
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить"
  />
  <div />
</div>
`;

exports[`company 3`] = `
<div>
  <div>
    <input
      onClick={[Function]}
      type="button"
      value="Velcom"
    />
    <input
      onClick={[Function]}
      type="button"
      value="MTC"
    />
    <h2>
      Компания: 
      Velcom
    </h2>
  </div>
  <div
    className="filter"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <div>
    <table>
      <tbody>
        <tr>
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
        <tr>
          <td>
            Иванов
          </td>
          <td>
            Иван
          </td>
          <td>
            Иванович
          </td>
          <td>
            200
          </td>
          <td
            style={
              Object {
                "backgroundColor": "green",
              }
            }
          >
            active
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr>
          <td>
            Сидоров
          </td>
          <td>
            Сидор
          </td>
          <td>
            Сидорович
          </td>
          <td>
            250
          </td>
          <td
            style={
              Object {
                "backgroundColor": "green",
              }
            }
          >
            active
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
        <tr>
          <td>
            Петров
          </td>
          <td>
            Петр
          </td>
          <td>
            Петрович
          </td>
          <td>
            180
          </td>
          <td
            style={
              Object {
                "backgroundColor": "green",
              }
            }
          >
            active
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить"
  />
  <div />
</div>
`;

exports[`company 4`] = `
<div>
  <div>
    <input
      onClick={[Function]}
      type="button"
      value="Velcom"
    />
    <input
      onClick={[Function]}
      type="button"
      value="MTC"
    />
    <h2>
      Компания: 
      Velcom
    </h2>
  </div>
  <div
    className="filter"
  >
    <input
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <div>
    <table>
      <tbody>
        <tr>
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
        <tr>
          <td>
            Григорьев
          </td>
          <td>
            Григорий
          </td>
          <td>
            Григорьевич
          </td>
          <td>
            -220
          </td>
          <td
            style={
              Object {
                "backgroundColor": "red",
              }
            }
          >
            blocked
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Редактировать"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="Удалить"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Добавить"
  />
  <div />
</div>
`;
